### PR TITLE
[GH-726] Add template_cookbook property

### LINKED ISSUE #726: Enable template_cookbook property in default apache2_install
## :ghost: Brief Description

It's impossible to use template_cookbook parameter (part of apache2_config) to inject apache2.conf.erb from my cookbook in default installation (apache2_install).

## :pancakes: Cookbook version

v8.6.0

## :woman_cook:  Chef-Infra Version

16.6.7

## :tophat: Platform details

RHEL7 (on-prem)

## Steps To Reproduce

Use
```ruby
apache2_install 'custom' do
  template_cookbook 'my_cookbook'
end
```

## :police_car: Expected behavior

Template httpd.conf/apache2.conf is provided from my_cookbook

## :heavy_plus_sign: Additional context

N/A

### PR DESCRIPTION
**Pull request by gagarinfan**
# Description

Provide template_cookbook property to enable passing own template for httpd.conf/apache2.conf in apache2_install

## Issues Resolved

#726 

## Check List

- [ ] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
@gagarinfan can you please update the resource documentation and also add a test case for this in one of the test recipes along with an InSpec test? Thanks!

**Comment by gagarinfan** [comment]
@ramereth Done

File: kitchen.yml [hunk 1]
```diff
@@ -24,29 +24,32 @@ platforms:
...
-      - recipe[test::php]
-  - name: pkg_name
-    run_list:
-      - recipe[test::pkg_name]
-    includes:
-      - centos-7
-  - name: mod_wsgi
-    run_list:
-      - recipe[test::wsgi]
+   - name: default
```
Comment (ramereth) [inline kitchen.yml]: @gagarinfan can you please fix the indentation issue?

**Review by ramereth — CHANGES_REQUESTED** [review]

**Review by Xorima — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [8.7.0](https://github.com/sous-chefs/apache2/releases/tag/8.7.0)

**Pull request merged**